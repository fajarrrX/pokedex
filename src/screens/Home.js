import React, { useState, useEffect } from "react";
import {
  View,
  Text,
  ScrollView,
  Image,
  TouchableOpacity,
  StyleSheet,
  TextInput,
  Alert,
} from "react-native";
import handleResponse from "../utils/handleResponseFetch";

export default function Home() {
  const [pokemons, setPokemons] = useState([]);
  const [searchfeild, setSearchfeild] = useState("");
  const [error, setError] = useState(null);

  useEffect(() => {
    fetchPokemons();
    return () => pokemons;
  }, []);

  const fetchPokemons = async () => {
    fetch("https://pokeapi.co/api/v2/pokemon?limit=500")
      .then(handleResponse)
      .then((pokemons) => setPokemons(pokemons.results))
      .catch((error) => setError(error));
  };
  return (
    <View>
      <View style={styles.searchContainer}>
        <TextInput
          style={styles.searchfeild}
          placeholder="Search Pokemons"
          onChangeText={(value) => setSearchfeild(value)}
          value={searchfeild}
        />
      </View>
      <ScrollView>
        <View style={styles.container}>
          {pokemons
            .filter((pokemon) =>
              pokemon.name.toLowerCase().includes(searchfeild.toLowerCase())
            )
            .map((pokemon, index) => {
              return (
                <TouchableOpacity
                  activeOpacity={0.5}
                  key={index}
                  style={styles.card}
                  onPress={() => Alert.alert(`Hi i'm ${pokemon.name}`)}
                >
                  <Image
                    style={{ width: 150, height: 150 }}
                    source={{
                      uri: `https://img.pokemondb.net/sprites/omega-ruby-alpha-sapphire/dex/normal/${pokemon.name}.png`,
                    }}
                  />
                  <Text>{pokemon.name}</Text>
                </TouchableOpacity>
              );
            })}
        </View>
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "row",
    flexWrap: "wrap",
    justifyContent: "center",
    marginTop: 50,
  },
  card: {
    alignItems: "center",
    borderWidth: 1,
    borderRadius: 20,
    borderColor: "#dbdbdb",
    marginHorizontal: 10,
    marginVertical: 10,
  },
  searchContainer: {
    backgroundColor: "#fff",
    position: "absolute",
    zIndex: 1,
    marginTop: 0,
    width: "100%",
  },
  searchfeild: {
    height: 40,
    borderWidth: 1,
    borderColor: "#dbdbdb",
    textAlign: "center",
    width: "100%",
    borderRadius: 10,
  },
});
