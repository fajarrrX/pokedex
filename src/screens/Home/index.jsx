import React from "react";
import { StyleSheet, Text, View, SafeAreaView, ScrollView } from "react-native";

export default function Home() {
  return (
    <SafeAreaView>
      <ScrollView>
        <Text>Lorem ipsum dolor sit amet consectetur adipisicing elit. Vero, ratione, laborum voluptate, nulla tempora et minima nesciunt porro assumenda sapiente delectus perspiciatis consectetur alias culpa! Et saepe, laboriosam numquam commodi quaerat repellat perferendis delectus libero tenetur nostrum, fugiat sapiente iure doloremque asperiores maxime quisquam tempora itaque dicta pariatur eos error. Debitis illum voluptatibus est ad accusantium? Ullam accusamus voluptatum, earum qui voluptatem esse error asperiores ducimus nemo ut ipsum quo eligendi ipsam maiores quas. Quidem, nihil nam aut minus, optio iure exercitationem deleniti unde id atque aperiam laudantium numquam eveniet beatae, maiores maxime. Exercitationem aliquid iste ipsa minima quisquam minus porro deserunt pariatur quis dolorum? Dignissimos expedita architecto perferendis possimus eaque adipisci quo repudiandae fugit, odit commodi voluptatibus eveniet iusto quas incidunt temporibus. Vitae fugit beatae, eum iusto cupiditate quibusdam possimus, culpa vero incidunt sequi impedit et qui ea deserunt natus amet nihil totam repudiandae necessitatibus animi illo eligendi! Accusantium laboriosam, ut dignissimos distinctio aspernatur nostrum inventore quos maiores rem. Ratione fugiat magnam quod omnis. Repudiandae sapiente recusandae illum consequatur. Quis repellendus earum ea dolorem. Nisi hic non accusantium nemo sed, fuga molestiae cumque debitis nesciunt facilis possimus doloribus, itaque perspiciatis numquam maxime unde tenetur, rerum optio? At vel doloribus, earum dolorum necessitatibus natus doloremque est dignissimos voluptates quidem minus veniam porro id laboriosam non dicta officia provident eligendi suscipit nobis ratione aut iusto ducimus! Minus, ut quis eos molestiae corporis nemo voluptatibus facere corrupti odit maxime architecto distinctio, libero modi ratione, at eaque cum! Esse, exercitationem? Possimus magnam deserunt sint libero, at nemo ab sequi sapiente amet ratione voluptatum tenetur. Provident ratione blanditiis earum odit rerum nostrum maiores hic explicabo nemo, voluptatibus sed rem, suscipit delectus, similique eveniet fugiat qui reiciendis quaerat. Necessitatibus eos laboriosam saepe sequi architecto neque quisquam error delectus deserunt ducimus! Non, sapiente temporibus? Sapiente aliquid itaque rerum modi obcaecati necessitatibus magnam beatae quis, incidunt totam iusto. Facere porro iste commodi reiciendis impedit! Inventore amet nostrum consequuntur cum explicabo, tenetur doloremque officia dolore sapiente ipsa, officiis doloribus soluta tempora iure magni voluptates ipsum aliquid perferendis reprehenderit porro ut mollitia molestiae! Eaque asperiores recusandae eveniet minima aperiam quisquam at, laboriosam molestiae maiores unde delectus facere fuga quasi tempore saepe dolorem corrupti velit consectetur ad neque, odio sed! Est nulla dolor quis saepe eveniet repudiandae deserunt, voluptatum praesentium at nostrum illum hic placeat nemo iure, assumenda magni maxime eum, pariatur iste maiores voluptas architecto debitis? Obcaecati vero optio odit consequuntur! Doloremque, odio! Repellat ipsum molestias unde nam consequuntur maxime repudiandae rerum. Suscipit facilis, maiores ab soluta omnis cumque. Provident officiis nihil maxime eum. Pariatur quas esse provident repellat harum voluptatum sequi, ratione nemo unde nesciunt amet delectus veritatis? Minus neque, laboriosam atque tempora, molestias eaque vitae debitis dignissimos soluta asperiores, ipsa magni? Ex assumenda illo cupiditate, maxime voluptate perferendis rem explicabo deserunt temporibus, error magni, eaque possimus aliquam asperiores dolorum commodi accusamus obcaecati! Deserunt possimus accusantium modi voluptates? Velit dolore optio, recusandae doloribus, voluptatem provident itaque laboriosam quia iusto possimus voluptate, similique unde explicabo et quod at quam.</Text>
      </ScrollView>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({});
