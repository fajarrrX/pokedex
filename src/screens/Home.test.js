import React from "react";
import renderer from "react-test-renderer";
import Home from "./Home";

it("Home screen renders correctly", () => {
  renderer.create(<Home />);
});

test("Home snapShot", () => {
  const snap = renderer.create(<Home />).toJSON();
  expect(snap).toMatchSnapshot();
});
