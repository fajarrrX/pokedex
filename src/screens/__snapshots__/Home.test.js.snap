// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapShot 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "marginTop": 0,
        "position": "absolute",
        "width": "100%",
        "zIndex": 1,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Search Pokemons"
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "#dbdbdb",
          "borderRadius": 10,
          "borderWidth": 1,
          "height": 40,
          "textAlign": "center",
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "center",
            "marginTop": 50,
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
